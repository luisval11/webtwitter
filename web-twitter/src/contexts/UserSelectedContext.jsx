import React from "react";

export const UserSelectedContext = React.createContext(null);
