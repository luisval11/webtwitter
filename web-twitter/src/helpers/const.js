export const DUMMY_POSTS = [
	{
		id: 1,
		authorId: 1,
		content: "Lorem ipsum content large",
		createdDate: new Date(2022, 2, 30, 14, 30),
	},
	{
		id: 2,
		authorId: 2,
		content: "Lorem ipsum content large 1",
		createdDate: new Date(2022, 2, 30, 10, 30),
	},
	{
		id: 3,
		authorId: 3,
		content: "Lorem ipsum content large 2",
		createdDate: new Date(2022, 2, 29, 10, 30),
	},
	{
		id: 4,
		authorId: 0,
		content: "Lorem ipsum content large 3",
		createdDate: new Date(2022, 2, 28, 10, 30),
	},
	{
		id: 5,
		authorId: 4,
		content: "Lorem ipsum content large 5",
		createdDate: new Date(2022, 2, 27, 10, 30),
	},
	{
		id: 6,
		authorId: 6,
		content: "Lorem ipsum content large 6",
		createdDate: new Date(2022, 2, 26, 9, 30),
	},
	{
		id: 7,
		authorId: 5,
		content: "Lorem ipsum content large 7",
		createdDate: new Date(2022, 2, 25, 12, 30),
	},
];

export const DUMMY_USERS = [
	{
		id: 1,
		name: "Alice",
		isFollowed: true,
	},
	{
		id: 2,
		name: "Bob",
		isFollowed: true,
	},
	{
		id: 3,
		name: "Mike",
		isFollowed: true,
	},
	{
		id: 4,
		name: "Beatriz",
		isFollowed: false,
	},
	{
		id: 5,
		name: "Ben",
		isFollowed: false,
	},
	{
		id: 6,
		name: "Jenny",
		isFollowed: false,
	},
];
