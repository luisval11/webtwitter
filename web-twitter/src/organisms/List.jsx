import React from "react";
import PropTypes from "prop-types";
import { ListGroup } from "react-bootstrap";
import { getEmptyListMessage } from "../helpers/utils/getEmptyListMessage";
import { ListItem } from "../molecules/listItem/ListItem";
import "./style.css";

export const List = ({ items, onClickItem, onClickAction, title, type }) => {
	return (
		<div className="list">
			{title && <h6>{title}</h6>}
			{items.length > 0 && (
				<ListGroup aria-labelledby={type} role="list">
					{items.map((item) => {
						return (
							<ListItem 
								key={item.id}
								type={type}
								item={item}
								onClickAction={onClickAction}
								onClickItem={onClickItem}
							/>
						);
					})}
				</ListGroup>
			)}
			{items.length === 0 && (
				<span>{getEmptyListMessage(title !== undefined ? title : null)}</span>
			)}
		</div>
	);
};

List.propTypes = {
	items: PropTypes.array.isRequired,
	title: PropTypes.string,
	type: PropTypes.string.isRequired,
	onClickAction: PropTypes.func,
	onClickItem: PropTypes.func,
};

export default List;
