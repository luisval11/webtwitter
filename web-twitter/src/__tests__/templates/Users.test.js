import { render, screen } from "@testing-library/react";
import { UserSelectedContext } from "../../contexts/UserSelectedContext";
import { DUMMY_USERS } from "../../helpers/const";
import List from "../../organisms/List";

describe("Test on Users", () => {
	const dummyUsers = DUMMY_USERS;
	const dummyFollowedUsers = dummyUsers.filter(
		(user) => user.isFollowed === true || user.isFollowed === null
	);
	const dummyUnfollowedUsers = dummyUsers.filter(
		(user) => user.isFollowed === false
	);

	test("should match initial snapshot for a List of users", () => {
		const handleClickItem = jest.fn();
		const handleClickAction = jest.fn();
		let wrapper = render(
			<UserSelectedContext.Provider value={{ selectedUser: null }}>
				<List 
					items={dummyFollowedUsers}
					onClickItem={handleClickItem}
					onClickAction={handleClickAction}
					type="user"
				/>
			</UserSelectedContext.Provider>
		);
		expect(wrapper).toMatchSnapshot();
	});
	//test("should have the number of items according to lenght of items props", () => {
	//	render(
	//		<UserSelectedContext.Provider value={{ selectedUser: null }}>
	//			<List
	//				items={dummyItems}
	//				onClickItem={() => console.log("Dummy onClickItem")}
	//				onClickAction={() => console.log("Dummy onClickAction")}
	//				type="user"
	//			/>
	//		</UserSelectedContext.Provider>
	//	);
	//	const dummyItemsLength = dummyItems.length;
	//	const itemsLength = screen.getAllByRole("listitem").length;
	//	expect(itemsLength).toBe(dummyItemsLength);
	//});
});
