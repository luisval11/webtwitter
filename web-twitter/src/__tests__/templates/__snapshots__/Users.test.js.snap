// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on Users should match initial snapshot for a List of users 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="list"
      >
        <div
          aria-labelledby="user"
          class="list-group"
          role="list"
        >
          <div
            class="list-item list-group-item"
            role="listitem"
          >
            <div
              class="user"
            >
              <span>
                Alice
              </span>
              <button
                class="btn btn-secondary"
                type="button"
              >
                Unfollow
              </button>
            </div>
          </div>
          <div
            class="list-item list-group-item"
            role="listitem"
          >
            <div
              class="user"
            >
              <span>
                Bob
              </span>
              <button
                class="btn btn-secondary"
                type="button"
              >
                Unfollow
              </button>
            </div>
          </div>
          <div
            class="list-item list-group-item"
            role="listitem"
          >
            <div
              class="user"
            >
              <span>
                Mike
              </span>
              <button
                class="btn btn-secondary"
                type="button"
              >
                Unfollow
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="list"
    >
      <div
        aria-labelledby="user"
        class="list-group"
        role="list"
      >
        <div
          class="list-item list-group-item"
          role="listitem"
        >
          <div
            class="user"
          >
            <span>
              Alice
            </span>
            <button
              class="btn btn-secondary"
              type="button"
            >
              Unfollow
            </button>
          </div>
        </div>
        <div
          class="list-item list-group-item"
          role="listitem"
        >
          <div
            class="user"
          >
            <span>
              Bob
            </span>
            <button
              class="btn btn-secondary"
              type="button"
            >
              Unfollow
            </button>
          </div>
        </div>
        <div
          class="list-item list-group-item"
          role="listitem"
        >
          <div
            class="user"
          >
            <span>
              Mike
            </span>
            <button
              class="btn btn-secondary"
              type="button"
            >
              Unfollow
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
